contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/SingleCountryCriterion.java

```diff
@@ -58,10 +58,10 @@ private boolean filterSubstation(Substation substation) {
             return false;
         }
         Country injectionCountry = substation.getCountry().orElse(null);
-        if (injectionCountry == null) {
+        if (injectionCountry == null && !countries.isEmpty()) {
```

Comment:
Is it linked to @etiennehomer PR? Maybe don't duplicate the modifications to prevent conflicts

Reply:
The  other PR is to be closed. Don't forget to consider the comments on the other PR.

Reply:
I added Objects.requireNonNull(countries);